### PR TITLE
UCS/BUILD: Support custom path for binutils dependency

### PR DESCRIPTION
**Pull request by yosefe**
## Why
Allow building UCX with a custom-installed binutils

### CODE REVIEW

**Comment by yosefe** [comment]
@dmitrygx @brminich can you pls review?
@artemry-nv FYI

**Comment by yosefe** [comment]
bot:pipe:retest

File: src/ucs/Makefile.am [hunk 1]
```diff
@@ -11,12 +11,15 @@ AUTOMAKE_OPTIONS    = nostdinc # avoid collision with built-in debug.h
 lib_LTLIBRARIES     = libucs.la
 bin_PROGRAMS        =
 
-libucs_la_CPPFLAGS = $(BASE_CPPFLAGS) -DUCX_MODULE_DIR=\"$(moduledir)\" \
+libucs_la_CPPFLAGS = $(BASE_CPPFLAGS) \
+                     $(BFD_CPPFLAGS) \
```
Comment (dmitrygx) [inline src/ucs/Makefile.am:15]: maybe move to the previous line?

File: src/ucs/Makefile.am [hunk 2]
```diff
@@ -11,12 +11,15 @@ AUTOMAKE_OPTIONS    = nostdinc # avoid collision with built-in debug.h
...
+                     -DUCX_MODULE_DIR=\"$(moduledir)\" \
                      -DUCX_CONFIG_DIR=\"$(ucx_config_dir)\"
-libucs_la_CFLAGS   = $(BASE_CFLAGS)
-libucs_la_LDFLAGS  = -ldl $(NUMA_LIBS) -version-info $(SOVERSION)
+libucs_la_CFLAGS   = $(BASE_CFLAGS) \
+                     $(BFD_CFLAGS)
```
Comment (dmitrygx) [inline src/ucs/Makefile.am:19]: maybe move to the previous line?

File: src/ucs/configure.m4 [hunk 3]
```diff
@@ -32,54 +32,102 @@ AM_CONDITIONAL([HAVE_PROFILING],[test "x$HAVE_PROFILING" = "xyes"])
...
+            [Enable using BFD support for detailed backtrace (default is guess).])],
+            [], [with_bfd=guess])
+AS_IF([test "x$with_bfd" != xno],
+      [
+       # Do not define BFD_CFLAGS, BFD_LIBS, etc to make sure automake will not
+       # try to use them when bfd_happy=no
+       BFD_CHECK_CFLAGS=""
+       BFD_CHECK_LIBS="-lbfd -ldl -lz"
+       AS_IF([test "x$with_bfd" = "xguess" -o "x$with_bfd" = "xyes"],
+             [BFD_CHECK_PPFLAGS=""
```
Comment (dmitrygx) [inline src/ucs/configure.m4:46]: `BFD_CHECK_CPPFLAGS`

File: src/ucs/configure.m4 [hunk 4]
```diff
@@ -32,54 +32,102 @@ AM_CONDITIONAL([HAVE_PROFILING],[test "x$HAVE_PROFILING" = "xyes"])
...
+       CFLAGS="$CFLAGS $BFD_CHECK_CFLAGS"
+       CPPFLAGS="$CPPFLAGS $BFD_CHECK_CPPFLAGS"
+       LDFLAGS="$LDFLAGS $BFD_CHECK_LDFLAGS"
+       LIBS="$LIBS $BFD_CHECK_LIBS"
+
+       bfd_happy="yes"
+       AC_CHECK_LIB(bfd, bfd_openr, [],
+                    [
+                     # If cannot link with bfd, try adding known dependency libs
+                     unset ac_cv_lib_bfd_bfd_openr # force re-check
```
Comment (dmitrygx) [inline src/ucs/configure.m4:66]:
```
# If cannot link with bfd, try adding known dependency libs
# unset 'ac_cv_lib_bfd_bfd_openr' and force re-check
```

**Review by brminich — APPROVED** [review]

**Pull request merged**